hip_data_tools/aws/athena.py

```diff
@@ -469,3 +471,82 @@ def get_athena_columns_from_dataframe(data_frame: DataFrame) -> List[dict]:
         {"column": field_name, "type": _PYTHON_TO_ATHENA_DATA_TYPE_MAP.get(field_type, "STRING")}
         for
         field_name, field_type in column_dtype.items()]
+
+
+@dataclass
+class AthenaTablePartitionsHandlerSettings:
+    """Athena table partitions handler settings"""
+    database: str
+    conn: AwsConnectionManager
+    output_bucket: str
+    output_key: str
+    table: str
+    s3_bucket: str
+    s3_key: str
+    partition_col_names: list
+    key_suffix: Optional[str]
+
+
+class AthenaTablePartitionsHandlerUtil(AthenaUtil):
+    """
+    Utility class for to handle athena partitions
+
+    Args:
+        settings (AthenaTablePartitionsHandlerSettings): athena table partitions handler settings
+    """
+
+    def __init__(self, settings: AthenaTablePartitionsHandlerSettings):
+        super().__init__(database=settings.database, conn=settings.conn,
+                         output_bucket=settings.output_bucket,
+                         output_key=settings.output_key)
+        self.table = settings.table
+        self.s3_bucket = settings.s3_bucket
+        self.s3_key = settings.s3_key
+        self.partition_col_names = settings.partition_col_names
+        self.key_suffix = settings.key_suffix
+
+    def add_partitions_as_chunks(self, number_of_partitions_per_chunk: int):
+        """
+        Add partitions as chunks
+        Args:
+            number_of_partitions_per_chunk (int): size of chunk
+        Returns: None
+
+        """
+        s3u = S3Util(conn=self.conn, bucket=self.s3_bucket)
+        key_list = s3u.get_keys(key_prefix=self.s3_key)
+        dir_path_list = self._get_dir_path_list(key_list)
+        list_of_dict = [self._get_partition_dict(key) for key in list(set(dir_path_list))]
+        chunked_list_of_dict = [list_of_dict[i:i + number_of_partitions_per_chunk] for i in
+                                range(0, len(list_of_dict), number_of_partitions_per_chunk)]
+        for chunk in chunked_list_of_dict:
+            add_partitions_query = self._get_add_partitions_query_for_chunk(chunk)
+            self.run_query(query_string=add_partitions_query)
+
+    def _get_dir_path_list(self, key_list):
```

Comment:
this could be a function instead of a class method

Reply:
fixed